examples/custom-node/src/evm/env.rs

```diff
@@ -0,0 +1,139 @@
+use crate::primitives::{CustomTransaction, CustomTransactionEnvelope, TxPayment};
+use alloy_eips::Typed2718;
+use alloy_evm::{FromRecoveredTx, FromTxWithEncoded};
+use alloy_primitives::{Address, Bytes, TxKind, B256, U256};
+use op_revm::OpTransaction;
+use reth_ethereum::evm::revm::context::TxEnv;
+
+pub struct CustomTxEnv(TxEnv);
+
+impl revm::context::Transaction for CustomTxEnv {
+    type AccessListItem<'a>
+        = <TxEnv as revm::context::Transaction>::AccessListItem<'a>
+    where
+        Self: 'a;
+    type Authorization<'a>
+        = <TxEnv as revm::context::Transaction>::Authorization<'a>
+    where
+        Self: 'a;
+
+    fn tx_type(&self) -> u8 {
+        self.0.tx_type()
+    }
+
+    fn caller(&self) -> Address {
+        self.0.caller()
+    }
+
+    fn gas_limit(&self) -> u64 {
+        self.0.gas_limit()
+    }
+
+    fn value(&self) -> U256 {
+        self.0.value()
+    }
+
+    fn input(&self) -> &Bytes {
+        self.0.input()
+    }
+
+    fn nonce(&self) -> u64 {
+        self.0.nonce()
+    }
+
+    fn kind(&self) -> TxKind {
+        self.0.kind()
+    }
+
+    fn chain_id(&self) -> Option<u64> {
+        self.0.chain_id()
+    }
+
+    fn gas_price(&self) -> u128 {
+        self.0.gas_price()
+    }
+
+    fn access_list(&self) -> Option<impl Iterator<Item = Self::AccessListItem<'_>>> {
+        self.0.access_list()
+    }
+
+    fn blob_versioned_hashes(&self) -> &[B256] {
+        self.0.blob_versioned_hashes()
+    }
+
+    fn max_fee_per_blob_gas(&self) -> u128 {
+        self.0.max_fee_per_blob_gas()
+    }
+
+    fn authorization_list_len(&self) -> usize {
+        self.0.authorization_list_len()
+    }
+
+    fn authorization_list(&self) -> impl Iterator<Item = Self::Authorization<'_>> {
+        self.0.authorization_list()
+    }
+
+    fn max_priority_fee_per_gas(&self) -> Option<u128> {
+        self.0.max_priority_fee_per_gas()
+    }
+}
+
+impl FromRecoveredTx<CustomTransaction> for CustomTxEnv {
+    fn from_recovered_tx(tx: &CustomTransaction, sender: Address) -> Self {
+        CustomTxEnv(match tx {
+            CustomTransaction::BuiltIn(tx) => {
+                OpTransaction::<TxEnv>::from_recovered_tx(tx, sender).base
+            }
+            CustomTransaction::Other(tx) => TxEnv::from_recovered_tx(tx, sender),
+        })
+    }
+}
+
+impl FromTxWithEncoded<CustomTransaction> for CustomTxEnv {
+    fn from_encoded_tx(tx: &CustomTransaction, sender: Address, encoded: Bytes) -> Self {
+        CustomTxEnv(match tx {
```

Comment:
it could be that we need to implement these directly to `Extended`, which I think we can also do
we could even move `Extended` to alloy-consensus once we're happy with it, then we can access it in alloy-evm, because maybe adding alloy-evm as dep to reth-primitives isn't what we want, or we can feature gate,
or we can make the alloy-evm traits leaner by extracting those.
a lot of options

Reply:
good thinking, thanks for sharing